### ISSUE TITLE
Rails 5 Support?

### ISSUE THREAD

**Issue by archonic** [RailsApps/rails-stripe-membership-saas#153]
Here's a nasty question. I have a big complex app I plan to build which I hope will involve rails-stripe-membership-saas but also involves a node server for push notifications and an API for an assortment of things. Rails 5 looks like it may emerge in Fall of 2015 and can bring these things in house with Active Cable and native API support.

Being API only is the kicker. So 2 questions, for Mr Kehoe but also the community:

1. Is support for Rails 5 planned?
2. Are folks interested in an API supporting or API only version of this application? That would require splitting the project into essentially 2 projects - the API side and the MV* frontend framework side with something like Backbone.

**Comment by DanielKehoe** [comment]
Yes we should develop an API only version of the application. I doubt we'll use the Payola gem for that. So likely a new application providing similar functionality.

**Comment by archonic** [comment]
On topic of payola, I have a PR that should get accepted soon which allows registration without a credit card on free plans and plans with a trial. I hope to have a PR here which implements those changes once payolas JS is refactored. I don't know enough about payola's implementation / API structure to know what would be needed to make that happen, but with or without it, an API version of this application is very exciting.

**Comment by kathyonu** [comment]
An API version of this app as rails-stripe-membership-api ?
Direct connections to Stripe, handling all of Stripe's communications, storing all required data and receiving and acting upon their webhooks  so it is a fully functional app ? 
That is my goal for this app. May I recommend that you give early notice of what needs to be learned to be able to follow and perhaps contribute to this new development of the rails-stripe-membership-saas app ?  This app is well worthy of such improvements, and to be running on Rails 5 when released, so count me in, and if I can help in any way, say the word.

**Comment by meganedesigns** [comment]
I would love to see an updated Rails 5 version of this project. Is that in the works?

**Comment by DanielKehoe** [comment]
I recommend using the https://github.com/RailsApps/rails-stripe-coupons example application. I may create a new version of this app by extending the rails-stripe-coupons example application. I am no longer happy with the Payola gem because of its complexity and inconsistent support.

**Comment by rssluca** [comment]
Hi Daniel!

Have you had a chance to extend the coupons example as you mentioned?

Thanks a lot for your work.

Luca

**Comment by DanielKehoe** [comment]
@rssluca Sorry to say but I haven't had much time to support continued development of the example applications. I previously was able to work full-time on the RailsApps project but sales of my books and tutorials have declined so I'm now working full time for companies. Happy to provide guidance if someone wants to contribute, though.

**Comment by kathyonu** [comment]
I see this thread began with Daniel Kehoe’s rail-stripe-membership-saas app, and bringing it forward to Rails 5 from its current 3.2 branch and now I see a desire to bring that app forward to Rails 5, sans the Payola gem, or remove Payola from the current master.  Daniel will know that is music to my ears, as we already had all Stripe functionality working and I had written over 200 tests on the 3.2 code flow. When I saw the update of the memberships app containing Payola .. I knew it was not the way to go, and yet, it was a done deal at that time. The 3.2 branch lay fallow .. so I carried on alone.  I have not kept my fork of the app up to date, working solely on the development and production of the clone.

I have brought that app forward to Rails 4.2.10 ..
I have attempted to bring it forward to Rails 5 five times, and fail each time.
I have attempted to bring it forward with use of enum, removing CanCanCan and Ability for Role based access management, and have failed, once.

I am an autodidactic programmer, and see the value of Daniel’s mentorship in these tutorials, his willingness to answer questions, when he is able to work Open Source. Without that mentorship, I have not been able re-write codes nor test my way through, with our in-production membership app in these matters, up to Rails 5, nor make some other key improvements.

I am interested in working with others to bring the rails-stripe-membership-saas app up to latest Rails 5. Or, taking Payola out of the current master and replacing with working codes .. seems daunting, and I do not know Payola, because I already knew Stripe and their codes directly, and I had written RSpec, stripe-ruby-mock tests on the app and Stripe live. Payola seemed to be to be an unnecessary intruder of sorts, another gem to depend on to stay current with Stripe.

If I can contribute to the updating of the 3.2 branch of the membership app, or the removal of Payola from the current master, or the contribution of tests to the app, I am all in. I can also help Daniel in the proof-reading of the updated tutorial, or I can work to update it myself or with others. Again, Daniel’s genius in writing the tutorial still has me in awe. Thank you, Daniel, for all you have done. I for one miss your professional guidances in these matters.
